marklogic-data-hub/src/test/java/com/marklogic/hub/flow/impl/FlowRunnerTest.java

```diff
@@ -247,6 +249,34 @@ public void testIngestCSVasXMLCustomDelimiter(){
         Assertions.assertEquals(count, 25);
     }
 
+    @Test
+    public void testSourceQueryLimit(){
+        Map<String,Object> opts = new HashMap<>();
+
+        runAsDataHubOperator();
+        RunFlowResponse resp = runFlow("testFlow", "4", UUID.randomUUID().toString(),opts, new HashMap<>());
+        flowRunner.awaitCompletion();
+        Assertions.assertTrue(getDocCount(HubConfig.DEFAULT_STAGING_NAME, "csv-tab-coll") == 25);
+        Assertions.assertTrue(JobStatus.FINISHED.toString().equalsIgnoreCase(resp.getJobStatus()));
+
+        opts.put("sourceQuery", "cts.collectionQuery('csv-tab-coll')");
+        opts.put("sourceQueryLimit", 2);
+        resp = runFlow("testFlow", "6", UUID.randomUUID().toString(), opts, new HashMap<>());
+        flowRunner.awaitCompletion();
+        Assertions.assertEquals(2, getDocCount(HubConfig.DEFAULT_FINAL_NAME, "xml-map"));
+        Assertions.assertTrue(JobStatus.FINISHED.toString().equalsIgnoreCase(resp.getJobStatus()));
```

Comment:
I would just do assertEquals("finished", resp.getJobStatus()) - we expect that exact string, we don't need an equalsIgnoreCase here.